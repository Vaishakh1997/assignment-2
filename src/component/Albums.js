import React, { Component } from 'react';
import '../App.css'
const axios = require('axios');

class Albums extends Component {
    state = { 
        albumsData:[]
     }

    goToPhotos = (albumId) => {
        this.props.history.push(`/albums/${albumId}`)
    }

    componentDidMount(){
        axios({
            method: 'get',
            url: `https://jsonplaceholder.typicode.com/albums`
          })
            .then(response=> this.setState({albumsData:response.data}))
            .catch(error=> console.error(error))
    }
    
    render() { 
        return ( 
            <React.Fragment>
                <div className="albums-list">
                    {this.state.albumsData.map(album=>{
                        return (<div className="album" key={album.id} onClick={()=>this.goToPhotos(album.id)}>
                            <h2>{album.title}</h2>
                        </div>)
                    })}
                </div>
            </React.Fragment>
         );
    }
}
 
export default Albums;