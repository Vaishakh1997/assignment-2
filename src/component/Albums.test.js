import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import {Albums} from './Albums';

const mockStore = configureMockStore();
const store = mockStore({});
const props= {loading: false, albums:[1,2], error:'error'}

let wrapped = shallow(
    <Provider store={store}>
        <Albums {...props}/>
    </Provider>
)


describe("Albums Component", () => {

    it("Albums should render without throwing an error", () => {
        expect(wrapped.debug()).toMatchSnapshot();
    });
    
    it('Expected props loading as false', () => {
        expect(wrapped.dive().props().loading).toEqual(props.loading)
    });

    it('Expected props albums as [1,2]', () => {
        expect(wrapped.dive().props().albums).toEqual(props.albums)
    });

    it('Expected props error as error message', () => {
        expect(wrapped.dive().props().error).toEqual(props.error)
    });
    
});