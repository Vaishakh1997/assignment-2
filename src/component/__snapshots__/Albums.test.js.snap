// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testpage Component should render without throwing an error 1`] = `ShallowWrapper {}`;
